games/guess-it/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+            State::WinnerSelection => "WinnerSelection",
+            State::Payout => "Payout",
+            State::Refunding => "Refunding",
+            State::Destroyed => "Destroyed",
+        };
+        write!(f, "{}", outcome)
+    }
+}
+
+#[derive(TypeId, Decode, Encode)]
+pub struct Game {
+    name: String,
+    state: State,
+    bet: String,
+    players: HashMap<NonFungibleKey, Player>,
+    winner: String,
+    last_roll: String,
+}
+impl fmt::Debug for Game {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let formatted = str::replace(&*format!("{:#}", Game::serialize(&self).as_str()), "\\", "");
+        write!(f, "{}", formatted)
+    }
+}
+impl Game {
+    pub fn new(players: Option<HashMap<NonFungibleKey, Player>>, name: String, bet: String) -> Self {
+        Self {
+            name,
+            state: State::AcceptingChallenger,
+            bet,
+            players: players.unwrap_or(HashMap::new()),
+            winner: NonFungibleKey::from(0).to_string(),
+            last_roll: 0.to_string(),
+        }
+    }
+    pub fn serialize(game: &Self) -> String {
+        let players: HashMap<String, Player> = game.players.clone().into_iter()
+            .map(|(k, p)| (k.to_string(), p))
+            .collect();
+
+        let serializable = GameSerialized {
+            name: game.name.clone(),
+            state: game.state.clone(),
+            players,
+            winner: game.winner.clone(),
+            bet: game.bet.clone(),
+            last_roll: game.last_roll.clone(),
+        };
+        unescape(serde_json::to_string(&serializable).unwrap_or("".to_string())
+            .as_mut_str())
+            .unwrap_or("".to_string())
+    }
+}
+
+#[derive(TypeId, Decode, Encode, Clone, Deserialize, Serialize)]
+pub struct GameSerialized {
+    pub name: String,
+    pub state: State,
+    pub bet: String,
+    pub players: HashMap<String, Player>,
+    pub winner: String,
+    pub last_roll: String,
+}
+impl fmt::Debug for GameSerialized {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let formatted = serde_json::to_string(&self).unwrap().replace("\\", "");
+        write!(f, "{}", formatted)
+    }
+}
+impl GameSerialized {
+    pub fn empty(players: Option<HashMap<String, Player>>) -> Self {
+        Self {
+            name: "".to_string(),
+            state: State::AcceptingChallenger,
+            bet: 0.to_string(),
+            players: players.unwrap_or(HashMap::new()),
+            winner: NonFungibleKey::from(0).to_string(),
+            last_roll: 0.to_string(),
+        }
+    }
+}
+
+#[derive(Debug, TypeId, Describe, Decode, Encode, Clone, Serialize, Deserialize)]
+pub struct Player {
+    guess: String,
+}
+impl Player {
+    pub fn empty() -> Player {
+        return Self {
+            guess: "".into(),
+        }
+    }
+
+    pub fn get_guess(&self) -> String { self.guess.clone() }
+    pub fn make_guess(&mut self, guess: String) { self.guess = guess; }
```

Comment:
The `guess` argument should not be a `String` here since this game is entirely based on numbers. A more appropriate type here would be `u8`, `u16`, `u32` or maybe `Decimal`. 

The use of a String means that somebody could put in a string which would later lead to undesired behavior happening.

Reply:
I was under the impression I'd tested this and thought only Strings could be interpreted from the resim command, but I'll change this going forward

Reply:
I think `guess` is the hash of the number the user is committing to bet on?